bundles/http/Citrix/SAP-Concur-Demo-HTTP/script.js

```diff
@@ -0,0 +1,603 @@
+  },
+  {
+    Type: 'Airline Fee',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/itinerary.82ae0728d114f607d5b618e3b4eba614.svg'
+  },
+  {
+    Type: 'Meal',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/change_request.b0ee4a80fb3ec8b889c043706b26bb1d.svg'
+  },
+  {
+    Type: 'Entertainment',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/default-broken.b3426980ecaefcfc0778d475c1fe4dd6.svg'
+  },
+  {
+    Type: 'Other',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/feedback.c533ed2d280d39d9288b2fd741a18651.svg'
+  },
+  {
+    Type: 'Office Supplies',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/case.2bb73f1b3eff52a8134fd591447bdaa1.svg'
+  },
+  {
+    Type: 'Advertising/Marketing',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/lead.e61e72321674c9469b115c0552c5dc65.svg'
+  },
+  {
+    Type: 'Office Software',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/report_update.2d453d1646665c2ede6de83f848a6566.svg'
+  }
+]
+
+async function syncReports ({ dataStore, integrationParameters }) {
+  console.log('Synching data...')
+
+  countTotalAmount()
+
+  dataStore.save('reports', reports)
+  dataStore.save('expenses', expenses)
+  dataStore.save('categories', categories)
+  console.log("Synchronization complete")
+}
+
+function countTotalAmount() {
+  let amounts = new Map()
+  expenses.forEach(e => {
+    amounts.set(e.ReportName, (amounts.has(e.ReportName)? amounts.get(e.ReportName) : 0) + e.Amount)
+  })
+  reports.forEach(r => r.TotalAmount = amounts.get(r.Id))
+}
+
+async function incrementalSynch({dataStore, client}) {
+  console.log("Running incremental synchronization...")
+  //pulling a random report from the report list
+  let report = JSON.parse(JSON.stringify(reports[Math.floor(Math.random() * (reports.length))]))
+  //this looks really scary, but we need to get the string value of report id, transfer it to int, add the randomly generated value...
+  const reportId = parseInt(report.Id) + Math.floor(Math.random() * (1000 - reports.length)) + reports.length
+  //and then return it back to being a string
+  let rid = reportId.toString()
+  report.Id = rid
+  report.DateSubmitted=date.toLocaleDateString()
+
+  //randomly generating a number of expenses for a report
+  let numOfExpenses = Math.floor(Math.random() * (7 - 1)) + 1
+  //expenses will be saved into the array 
+  let iExpenses = []
+  let totalamount = 0
+  //pulling the last id of an expense so we can adequately increment the ids
+  //we have to slice the value since the ids are far outside of integer range 
+  let eLastId = parseInt(expenses[expenses.length-1].Id.slice(12,17))
+  for (let i=0; i< numOfExpenses; i++) {
```

Comment:
replace `for` with `map` and try to avoid modifying variables or globals outside the block.
```
const expenses = _.range(0, numOfExpenses).map(i => {
  const expense = getRandomExpense(expensesTemplate)
  expense.id = ....
  expense .TransactionDate = ...
  // etc
  return expense
})
```

for instace: removing `totalAmount`, see comment with `calculateReportsWithTotals` - it's the same approach



```
dataStore.save('reports', reportWithTotalAmount(report))
```

Reply:
Resolved in the new version of the script